pkg/threescale/threescale.go

```diff
@@ -277,6 +274,21 @@ func errorToRpcStatus(err error) func(string) rpc.Status {
 	}
 }
 
+func backendResponseToRpcStatus(result *authorizer.BackendResponse) func(string) rpc.Status {
+	if result == nil || result.RawResponse == nil {
+		return status.WithUnknown
+	}
+
+	if val, ok := result.RawResponse.(*http.Response); ok {
+		code, ok := httpStatusToRpcStatus[val.StatusCode]
+		if !ok {
+			return status.WithUnknown
+		}
+		return code
+	}
+	return status.WithUnknown
```

Comment:
Quite a lot of return statements here which _maybe_ could be simplified via assigninig `code` at the top with `status.WithUnknown` and rewriting the branching so it filters out the unhappy paths to end up with the matching map value. Just a suggestion for easier reading. Wdyt?